### PR TITLE
[maistra-2.6] Rebase commits between 2.5.0 and 2.5.2

### PR DESCRIPTION
**Pull request by yannuil**

### CODE REVIEW

**Review by jewertow — DISMISSED** [review]
OSSM-5556 was reverted: https://github.com/maistra/istio/pull/964

**Comment by yannuil** [comment]
@jewertow as you suggest in the reverted commit, we don't want to include the fix in the Z-stream. This is a Y-stream. Should we include it?

**Comment by jewertow** [comment]
No, we do not want that change at all.

**Comment by yannuil** [comment]
I have dropped a [commit](https://github.com/maistra/istio/commit/bd3d6530e48034b35236392de447a86d852013b1) to resolve the conflict with #1002. The commit was picked from istio/istio#48056, which includes some changes on [`pilot/pkg/config/kube/gateway/conditions.go`](https://github.com/istio/istio/pull/48056/files#diff-62af8b3a3ade7a9e214848a880e4a3d588214d5ccbfc7ff2ee0c477f467c18dc). Do we need to incorporate them as well? @jewertow @eoinfennessy

**Comment by eoinfennessy** [comment]
> I have dropped a [commit](https://github.com/maistra/istio/commit/bd3d6530e48034b35236392de447a86d852013b1) to resolve the conflict with #1002. The commit was picked from [istio/istio#48056](https://github.com/istio/istio/pull/48056), which includes some changes on [`pilot/pkg/config/kube/gateway/conditions.go`](https://github.com/istio/istio/pull/48056/files#diff-62af8b3a3ade7a9e214848a880e4a3d588214d5ccbfc7ff2ee0c477f467c18dc). Do we need to incorporate them as well?

Yes, I think we should incorporate the changes in `conditions.go`. You may be able to use the `controllerName` variable defined in `model.go`.

File: pilot/pkg/config/kube/gateway/controller_test.go [hunk 1]
```diff
@@ -43,7 +44,7 @@ import (
 
 var (
 	gatewayClassSpec = &k8s.GatewayClassSpec{
-		ControllerName: constants.ManagedGatewayController,
+		ControllerName: k8sv1.GatewayController(features.ManagedGatewayController),
```
Comment (eoinfennessy) [inline pilot/pkg/config/kube/gateway/controller_test.go]: For consistency, should we use `controllerName` here?

File: pilot/pkg/config/kube/gateway/deploymentcontroller_test.go [hunk 2]
```diff
@@ -62,7 +61,7 @@ func TestConfigureIstioGateway(t *testing.T) {
 			Name: "custom",
 		},
 		Spec: v1beta1.GatewayClassSpec{
-			ControllerName: constants.ManagedGatewayController,
+			ControllerName: k8sv1.GatewayController(features.ManagedGatewayController),
```
Comment (eoinfennessy) [inline pilot/pkg/config/kube/gateway/deploymentcontroller_test.go]: > For consistency, should we use `controllerName` here?

...and here also?

**Comment by eoinfennessy** [comment]
Do we want `tide/merge-method-rebase` on this PR?

**Review by eoinfennessy — APPROVED** [review]

**Comment by jewertow** [comment]
@eoinfennessy yes, we do.

**Pull request merged**